cpp/src/aztec/proof_system/work_queue/work_queue.cpp

```diff
@@ -50,14 +50,28 @@ size_t work_queue::get_scalar_multiplication_size(const size_t work_item_number)
     for (const auto& item : work_item_queue) {
         if (item.work_type == WorkType::SCALAR_MULTIPLICATION) {
             if (count == work_item_number) {
-                return item.constant == MSMSize::N ? key->circuit_size : key->circuit_size + 1;
+                return (item.constant == MSMType::MONOMIAL_N_PLUS_ONE) ? key->circuit_size + 1 : key->circuit_size;
             }
             ++count;
         }
     }
     return 0;
 }
 
+bool work_queue::get_scalar_multiplication_type(const size_t work_item_number) const
```

Comment:
A comment here (or in header) would be helpful since since it's not immediately obvious which return value corresponds to which SM type

Reply:
Done, thank you!